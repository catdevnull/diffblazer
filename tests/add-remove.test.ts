import { describe, it, expect } from 'vitest'
import HtmlDiff from '../src/lib/Diff'
import { minify } from 'html-minifier'
function html(source: TemplateStringsArray) {
	return minify(source.join(''), { collapseWhitespace: true, decodeEntities: true, html5: true }).replace(
		/>\s+</gm,
		'><',
	)
}

describe('Adding tags', function () {
	it('should be reported when adding new tags in the middle', function () {
		var html1 = html`<div><b>1</b> <strong>2</strong></div>`
		var html2 = html`<div><b>1</b> <i>new</i> <strong>2</strong></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})

	it('should be reported when adding new tags at the end', function () {
		var html1 = html`<div><b>1</b> <strong>2</strong></div>`
		var html2 = html`<div><b>1</b> <strong>2</strong> <i>new</i></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})

	it('should be reported when adding new tags at the start', function () {
		var html1 = html`<div><b>1</b> <strong>2</strong></div>`
		var html2 = html`<div><i>new</i> <b>1</b> <strong>2</strong></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})
})

describe('Removing tags', function () {
	it('should be reported when removing tags in the middle', function () {
		var html1 = html`<div><b>1</b> <i>removed</i> <strong>2</strong></div>`
		var html2 = html`<div><b>1</b> <strong>2</strong></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})

	it('should be reported when removing tags at the end', function () {
		var html1 = html`<div><b>1</b> <strong>2</strong> <i>removed</i></div>`
		var html2 = html`<div><b>1</b> <strong>2</strong></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})

	it('should be reported when removing tags at the start', function () {
		var html1 = html`<div><i>removed</i> <b>1</b> <strong>2</strong></div>`
		var html2 = html`<div><b>1</b> <strong>2</strong></div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})
})

describe('Multiple additions/removals', function () {
	it('should be correctly recognized', function () {
		var html1 = html`<div id="multi">
			<b>1</b> <br />
			<i>2</i>
			<p>3</p>
			<b>4</b> <a>5</a>
		</div>`
		var html2 = html`<div id="multi">
			<a>added</a> <b>1</b> <i>2</i> <strong>Hello</strong> <em>Hi!</em>
			<p>3</p>
			<b>4</b> <a>5</a> <em>Done.</em>
		</div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})

	it('multiple paragraphs', function () {
		var html1 = html`<div id="multi">
			<b>1</b> <br />
			<i>2</i>
			<p>3</p>
			<b>4</b> <a>5</a>
		</div>`
		var html2 = html`<div id="multi">
			<a>added</a> <b>1</b> <i>2</i> <strong>Hello</strong> <em>Hi!</em>
			<p>3</p>
			<b>4</b> <a>5</a> <em>Done.</em>
		</div>`
		var d = HtmlDiff.execute(html1, html2)
		expect(d).toMatchSnapshot()
	})
})
