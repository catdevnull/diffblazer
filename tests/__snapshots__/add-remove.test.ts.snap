// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Adding tags > should be reported when adding new tags at the end 1`] = `"<div><b>1</b><strong>2</strong><i><ins class='mod'><ins class='diffins'>new</ins></ins></i></div>"`;

exports[`Adding tags > should be reported when adding new tags at the start 1`] = `"<div><i><ins class='mod'><ins class='diffins'>new</ins></ins></i><b>1</b><strong>2</strong></div>"`;

exports[`Adding tags > should be reported when adding new tags in the middle 1`] = `"<div><b>1</b><i><ins class='mod'><ins class='diffins'>new</ins></ins></i><strong>2</strong></div>"`;

exports[`Multiple additions/removals > multiple paragraphs 1`] = `"<div id="multi"><a><ins class='diffins'>added</ins></a><b>1</b><br><i>2</i><strong><ins class='mod'><ins class='diffins'>Hello</ins></strong><em><ins class='diffins'>Hi!</ins></em><p>3</p><b>4</b><a>5</a><em><ins class='mod'><ins class='diffins'>Done.</ins></ins></em></div>"`;

exports[`Multiple additions/removals > should be correctly recognized 1`] = `"<div id="multi"><a><ins class='diffins'>added</ins></a><b>1</b><br><i>2</i><strong><ins class='mod'><ins class='diffins'>Hello</ins></strong><em><ins class='diffins'>Hi!</ins></em><p>3</p><b>4</b><a>5</a><em><ins class='mod'><ins class='diffins'>Done.</ins></ins></em></div>"`;

exports[`Removing tags > should be reported when removing tags at the end 1`] = `"<div><b>1</b><strong>2</strong><ins class='mod'><del class='diffdel'>removed</del></ins></div>"`;

exports[`Removing tags > should be reported when removing tags at the start 1`] = `"<div><ins class='mod'><del class='diffdel'>removed</del></ins><b>1</b><strong>2</strong></div>"`;

exports[`Removing tags > should be reported when removing tags in the middle 1`] = `"<div><b>1</b><ins class='mod'><del class='diffdel'>removed</del></ins><strong>2</strong></div>"`;
